crates/sp1/src/guest.rs

```diff
@@ -0,0 +1,51 @@
+//! This module implements the `ZkvmGuest` trait for the SP1 VM.
+use borsh::{BorshDeserialize, BorshSerialize};
+use sov_rollup_interface::zk::{Zkvm, ZkvmGuest};
+use sp1_zkvm::io;
+
+/// A guest for the SP1 VM. Implements the `ZkvmGuest` trait
+///  in terms of SP1's io::read and io::write functions.
+#[derive(Default)]
+pub struct SP1Guest {}
+
+impl SP1Guest {
+    /// Constructs a new SP1Guest
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Zkvm for SP1Guest {
+    #[cfg(feature = "native")]
```

Comment:
i don't think you need native flag here

Reply:
VerifyingKey is a wrapper of SP1VerifyingKey which exists under sp1-sdk. And when sp1-sdk is imported into the guest code, very very very bad things happen. This is going to be something we have to resolve to do recursive verification, because we need VK for verification.